### PR TITLE
Fixed #26827 -- Improved ModelState error message when relations refer model classes.

### PR DESCRIPTION
**Pull request by prashantpandey9**
- Updated value error message as per the solution given in the [ticket](https://code.djangoproject.com/ticket/26827)

### CODE REVIEW

File: django/db/migrations/state.py [hunk 1]
```diff
@@ -738,13 +738,17 @@ def __init__(
             # Sanity-check that relation fields are NOT referring to a model class.
             if field.is_relation and hasattr(field.related_model, "_meta"):
                 raise ValueError(
-                    'ModelState.fields cannot refer to a model class - "%s.to" does. '
-                    "Use a string reference instead." % name
+                    'Model fields in "ModelState.fields" cannot refer to a model class'
+                    ' - "%s.%s.%s.to" does. '
+                    "Use a string reference instead."
+                    % (self.app_label, self.name, name)
```
Comment (felixxm) [inline django/db/migrations/state.py:744]: Use f-strings
```suggestion
                    'Model fields in "ModelState.fields" cannot refer to a model class'
                    f' - "{self.app_label}.{self.name}.{name}.to" does. Use a string '
                    "reference instead."
```

File: django/db/migrations/state.py [hunk 2]
```diff
@@ -738,13 +738,17 @@ def __init__(
...
                 )
             if field.many_to_many and hasattr(field.remote_field.through, "_meta"):
                 raise ValueError(
-                    'ModelState.fields cannot refer to a model class - "%s.through" '
-                    "does. Use a string reference instead." % name
+                    'Model fields in "ModelState.fields" cannot refer to a model class '
+                    '- "%s.%s.%s.through" '
+                    "does. Use a string reference instead."
+                    % (self.app_label, self.name, name)
```
Comment (felixxm) [inline django/db/migrations/state.py:751]: Use f-strings
```suggestion
                    'Model fields in "ModelState.fields" cannot refer to a model class '
                    f'- "{self.app_label}.{self.name}.{name}.through" does. Use a '
                    "string reference instead."
```

File: tests/migrations/test_state.py [hunk 3]
```diff
@@ -1651,7 +1651,8 @@ def test_sanity_check_to(self):
         field = models.ForeignKey(UnicodeModel, models.CASCADE)
         with self.assertRaisesMessage(
             ValueError,
-            'ModelState.fields cannot refer to a model class - "field.to" does. '
+            'Model fields in "ModelState.fields" cannot refer to a model class - '
+            '"app.Model.field.to" does. '
```
Comment (felixxm) [inline tests/migrations/test_state.py:1655]: Wrap at 88 chars:
```suggestion
            'Model fields in "ModelState.fields" cannot refer to a model class'
            ' - "app.Model.field.through" does. Use a string reference instead.',
```

**Review by felixxm — COMMENTED** [review]
@prashantpandey9 Thanks :+1: I left minor comments.

**Comment by prashantpandey9** [comment]
> @prashantpandey9 Thanks 👍 I left minor comments.

Will do that. Thanks

**Comment by prashantpandey9** [comment]
@felixxm Hi, Thanks for the review. I have updated the changes according to the suggestions.

**Comment by felixxm** [comment]
@prashantpandey9 Thanks :+1: Welcome aboard :boat:

**Pull request merged**